Fix incorrect usage of ItemGroup in the Image block filters panel.

afercia opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->
See #67425

## What?
<!-- In a few words, what is the PR actually doing? -->
The `<ItemGroup>` component should only contain `<Item>` sub-components as it's intended to render a `list` with `listitems`. This quick PR aims to fix one incorrect usage.

It also contains a fix for the `Item` height new default of 40 pixels when it is rendered as a Button. This part is mainly for demoing a potential way to fix the size but I'm not sure it should be addressed this way. After crafting this PR I discovered the size issue is tracked in https://github.com/WordPress/gutenberg/issues/46741 and should be better addressed there. I'm glad to revert the size fix in this PR, just submitting as is so far, pending feedback.
Cc @WordPress/gutenberg-components 

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Components should render semantic markup.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
Uses the missing `Item` sub-component, as a Button.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->
- Select an Image block.
- In the block settings panel, switch to the 'Styles' tab.
- Observe hte 'Duotone' control.
- Inspect the source and observe the control is a button element, wrapped within a div element with `role="listitem"` which in turn is wrapped within a div element with `role="list"`.

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

## Screenshots or screencast <!-- if applicable -->

![Screenshot 2024-12-03 at 10 16 45](https://github.com/user-attachments/assets/14c83301-ff80-4445-9728-d1f53e546d0b)


<!-- If you would like to upload screenshots, feel free to use the table below when it is useful to show the difference between before and after the change. -->

|Before|After|
|-|-|
|<!-- Before screenshot here -->|<!-- After screenshot here -->|

afercia committed:
Fix incorrect usage of ItemGroup in the filters panel.

ciampo reviewed (commented):
> It also contains a fix for the Item height new default of 40 pixels when it is rendered as a Button. This part is mainly for demoing a potential way to fix the size but I'm not sure it should be addressed this way. After crafting this PR I discovered the size issue is tracked in https://github.com/WordPress/gutenberg/issues/46741 and should be better addressed there. I'm glad to revert the size fix in this PR, just submitting as is so far, pending feedback.

I would personally leave the height changes for another PR. The `ItemGroup` component doesn't currently set any height at all, and I would prefer to keep it that way for the time being.

We should discuss this aspect in https://github.com/WordPress/gutenberg/issues/46741, and potentially even decide to apply the sizing changes only when implementing the re-design discussed in #64445.

I wonder if we could do something like this (cc @mirka ):

```tsx
const CustomButtomItem = ( props ) => (
	<Button __next40pxDefaultSize { ...props } />
);

// Later in code
<Item as={ CustomButtomItem } { ...toggleProps }>
```

mirka commented on packages/block-editor/src/components/global-styles/filters-panel.js:
```diff
@@ -189,15 +189,16 @@ export default function FiltersPanel( {
-									<Button
+									<Item
+										as="button"
 										__next40pxDefaultSize
```
Does it even need this 40px size handling anyway? I'm seeing that the button is already pretty much the correct size (40px minus border widths = 38px) due to [these styles](https://github.com/WordPress/gutenberg/blob/1418350eb5a1f15e109fc96af385bdd029fc7304/packages/block-editor/src/components/global-styles/style.scss#L30). I might be missing something.

```suggestion
```

afercia replied:
Yes I noticed it's 38px + the borders. It's incorrect anyways.
I have no strong opinions, I can remove the `__next40pxDefaultSize` tempative fix, as I mentioned it was for demoing a potential way to fix it. I'd like to prevent designers complain about an incorrect height :)

afercia commented:
As an aside: if some designs want buttons with a grey border, then it should not be implemented with local overrides or incorrect usage of components such as by using `ItemGroup` only to draw the grey border.

Buttons should only use the variants of the base button component. If a new vaariant is desired, it should be considered t obe implemented in the base component.

afercia committed:
Revert change for __next40pxDefaultSize.

afercia commented:
Rebased and reverted the change related to the `__next40pxDefaultSize`. The button has now a height of 38 pixels:

![Screenshot 2024-12-09 at 12 57 01](https://github.com/user-attachments/assets/491c768f-5d3c-49ab-85d6-79774d7af5c9)

afercia committed:
Restore empty line.

tyxla reviewed (commented):
This looks good overall 👍 

One little thing I noticed that might need some work is the weird double bottom border we're getting sometimes:

![Screenshot 2024-12-09 at 15 15 52](https://github.com/user-attachments/assets/a9d83616-2f14-4d2e-856c-d040964faca1)


To repro this, I do the following:
* Click, but don't release the "Duotone" item
* Drag the mouse outside
* Drop anywhere

In this PR, I see the double border. On trunk, the item remains focused. I think keeping it focused makes sense.

afercia commented:
@tyxla thanks for your feedback.

> In this PR, I see the double border. On trunk, the item remains focused. I think keeping it focused makes sense.

The difference you are seeing in the focus indication comes from the fact that in this PR the `ItemGroup` is rendered as a button element and styled via `unstyledButton`, which uses [:focus-visible for the focus style](https://github.com/WordPress/gutenberg/blob/f6d8c69efa9ebd7c3dd1fdd02cddde779c970319/packages/components/src/item-group/styles.ts#L31-L42).

Instead, on trunk it's a Button component, which uses [:focus](https://github.com/WordPress/gutenberg/blob/f6d8c69efa9ebd7c3dd1fdd02cddde779c970319/packages/components/src/button/style.scss#L39-L44). 

`:focus-visible` is supposed to be shown only when using the keyboard, while `:focus` is shown also when using the mouse (as such, on trunk the focus indication is shown also when clicking with the mouse and persists when dragging and releasing the mouse outside of the button). 

It's an inconsistency in the focus indication between the Button component and ItemGroup as a button element which is outside of the scope of this issue. I think this has been implemented intentionally, but I'm not sure it's a good thinkg. I'll create a separate issue to consider making this behavior consistent.

tyxla reviewed (approved):
Works well and makes sense 👍 

Happy to get this one shipped considering #67747 will address my concern separately.

Thanks @afercia 🚀

afercia commented:
Thanks @tyxla for your review.

afercia commented:
@WordPress/gutenberg-core I guess it's okay to merge despite the two native tests being canceled because of the macOS deprecation thing?

afercia commented:
> Yeah, those have been failing on all PRs recently.

Yes, I read the conversation on Slack, just wondering whether a fix was coming soon. Will merge. Thanks.

tyxla commented:
Not really a fix, but we're skipping them temporarily in https://github.com/WordPress/gutenberg/pull/67799
